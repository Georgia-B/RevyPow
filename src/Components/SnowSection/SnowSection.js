import React from 'react';

const SnowSection = (props) => {
    if (!props.data) {
        return null;
    }
    return (
        <div className={props.className ? `snowsection ${props.className}` : "snowsection"}>
            <h3 className="snowsection__title">{props.data.title}</h3>
            <span className="snowsection__value">{props.data.value}{!props.noUnit && " cm"}</span>
        </div>
    );
}

export default SnowSection;
