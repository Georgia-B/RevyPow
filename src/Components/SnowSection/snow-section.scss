.snowsection {
  width: 30%;
  min-width: 120px;
  margin: 10px;
}

.snowsection__title {
  font-size: 20px;
  margin: 0 0 10px 0;
}
  
.snowsection__value {
  font-size: 32px;
}