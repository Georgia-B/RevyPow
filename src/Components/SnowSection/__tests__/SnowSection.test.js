import React from 'react'
import { cleanup, render } from '@testing-library/react'
import SnowSection from '../SnowSection';

describe('Components | <SnowSection /> with data', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<SnowSection data={{ title: "Base", value: "15" }} />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})

describe('Components | <SnowSection /> with no data', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<SnowSection />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})