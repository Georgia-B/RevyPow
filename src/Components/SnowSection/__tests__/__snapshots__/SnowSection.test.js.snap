// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | <SnowSection /> with data should render correctly 1`] = `
<DocumentFragment>
  <div
    class="snowsection"
  >
    <h3
      class="snowsection__title"
    >
      Base
    </h3>
    <span
      class="snowsection__value"
    >
      15 cm
    </span>
  </div>
</DocumentFragment>
`;

exports[`Components | <SnowSection /> with no data should render correctly 1`] = `<DocumentFragment />`;
